pkg/reconciler/knativeeventing/common/replicasenvvarstransform.go

```diff
@@ -53,13 +61,26 @@ func ReplicasEnvVarsTransform(client unstructuredGetter) mf.Transformer {
 			// Keep the existing number of replicas in the cluster for the deployment
 			apply.Spec.Replicas = current.Spec.Replicas
 
+			// Preserve the env vars in the existing cluster
 			for index := range current.Spec.Template.Spec.Containers {
 				currentContainer := current.Spec.Template.Spec.Containers[index]
-				for j := range apply.Spec.Template.Spec.Containers {
-					applyContainer := &apply.Spec.Template.Spec.Containers[j]
-					if currentContainer.Name == applyContainer.Name {
-						applyContainer.Env = currentContainer.Env
+				found, containerIndex := nameExistsContainers(currentContainer.Name, apply.Spec.Template.Spec.Containers)
+				if found {
+					applyContainer := &apply.Spec.Template.Spec.Containers[containerIndex]
+					mergedEnv := currentContainer.Env
+					for _, env := range applyContainer.Env {
+						found, envIndex := nameExistsEnvVars(env.Name, mergedEnv)
+						if !found {
+							// Add the new env var into the existing env vars, if it is not available in the existing
+							// cluster.
+							mergedEnv = append(mergedEnv, env)
+						} else if !nameExistsEnvVarNames(env.Name, envVarNames) {
+							// Set the env var value, if it is available in the existing
+							// cluster, but it is not in the preserved list of the env vars.
+							mergedEnv[envIndex] = env
+						}
```

Comment:
Should we be handling deletes of env vars that are no longer in the manifest too?

Reply:
Good question, I need to ask @lionelvillard or @matzew

Reply:
That was a rhetoric question. I think we should handle that :)

Reply:
> Should we be handling deletes of env vars that are no longer in the manifest too?

yes

Reply:
We have added the code to handle the deletion.